hkmc2/shared/src/test/mlscript/lifter/Loops.mls

```diff
@@ -43,29 +45,47 @@ fun foo() =
     set x += 1
     return () => x
 //│ JS (unsanitized):
-//│ let foo2, lambda2, lambda$2;
-//│ lambda$2 = function lambda$(x) {
-//│   return x
+//│ let foo2, lambda2, while$1, lambda$2, foo$capture5;
+//│ lambda$2 = function lambda$(foo$capture6) {
+//│   return foo$capture6.x$capture$0
 //│ };
-//│ lambda2 = (undefined, function (x) {
+//│ lambda2 = (undefined, function (foo$capture6) {
 //│   return () => {
-//│     return lambda$2(x)
+//│     return lambda$2(foo$capture6)
 //│   }
 //│ });
-//│ foo2 = function foo() {
-//│   let x, scrut, tmp2, tmp3, lambda$here;
-//│   x = 1;
-//│   tmp4: while (true) {
-//│     scrut = true;
-//│     if (scrut === true) {
-//│       tmp2 = x + 1;
-//│       x = tmp2;
-//│       lambda$here = runtime.safeCall(lambda2(x));
-//│       return lambda$here
-//│     } else { tmp3 = runtime.Unit; }
-//│     break;
+//│ while$1 = function while$(foo$capture6) {
+//│   let scrut, tmp2, lambda$here;
+//│   scrut = true;
+//│   if (scrut === true) {
+//│     tmp2 = foo$capture6.x$capture$0 + 1;
+//│     foo$capture6.x$capture$0 = tmp2;
+//│     lambda$here = runtime.safeCall(lambda2(foo$capture6));
+//│     return lambda$here
+//│   } else {
+//│     foo$capture6.tmp$capture$1 = runtime.Unit;
```

Comment:
I just noticed this crazy capture of a temporary variable that just stores the unit value. It's pretty crazy. I wonder why Unit is bound to a temporary variable in the first place.